### PR TITLE
fix: tests

### PR DESCRIPTION
**Pull request by rkostrzewski**
This adds:
- `npm install` done by `preact create` in tests for PRs and `master` branch
- `npm install --save-dev ../../path-to-preact-cli-under-test` in above tests so that preact-cli in node_modules is the one under tests (i.e. from branch)
- `.npmignore` which is kinda needed not to install all tests with their output folders when doing above
- operation logging to tests - should be easier to determine what failed (only on PRs and `master`)
- `prebuild` script which removes `./lib` folder with output (might run into some issues with this in the future)
- `--git` switch to `preact create` that is true by default not to create git repository inside another one
- tests now use `jest` and run in parallel (brings down time a bit)

Tests are looking stable.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -81,8 +82,6 @@
     "html-looks-like": "^1.0.2",
     "lodash": "^4.17.4",
     "ncp": "^2.0.0",
-    "node-sass": "^4.5.3",
```
Comment (rkostrzewski) [inline package.json]: I've added them for sass tests - no longer needed.

Doesn't matter now coz there is a bug in npm 5 which causes "devDepdencies" to be included when installing locally.

File: .npmignore [hunk 2]
```diff
@@ -0,0 +1,5 @@
+tests/
```
Comment (rkostrzewski) [inline .npmignore]: Ignored when using local dependency on npm 5 :cry:

File: .travis.yml [hunk 3]
```diff
@@ -8,15 +8,17 @@ addons:
...
 cache:
   npm: true
+  directories:
+    - $HOME/.npm
+    - ./node_modules
 
 before_script:
 - export LIGHTHOUSE_CHROMIUM_PATH=`which google-chrome-stable`
-- google-chrome-stable --headless --disable-gpu --remote-debugging-port=9222 http://localhost &
+- if [ "$TRAVIS_BRANCH" = "master" ]; then export WITH_INSTALL=true; export WITH_LOG=true; fi
```
Comment (rkostrzewski) [inline .travis.yml]: only run tests with `preact create --install` on master (+ PRs)

File: .npmignore [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+src/
+examples/
```
Comment (lukeed) [inline .npmignore]: Gotta keep this 💃

File: .travis.yml [hunk 5]
```diff
@@ -9,13 +9,17 @@ language: node_js
...
```
Comment (lukeed) [inline .travis.yml]: This is awesome! Didn't know you could control Travis to this extent.

File: src/commands/create.js [hunk 6]
```diff
@@ -179,7 +186,10 @@ async function initializeVersionControl(target) {
 	let git;
 	try {
 		git = await promisify(which)('git');
-	} catch (e) {}
+	} catch (e) {
+		process.stderr.write('Could not find git in $PATH.\n');
+		process.stdout.write('Continuing without initializing version control...\n');
```
Comment (lukeed) [inline src/commands/create.js]: Nice 👍

File: .travis.yml (hunk 5 above)
Comment (rkostrzewski) [inline .travis.yml]: Yeah, I love travis because it's so easy to work with, yet so powerful.

File: .npmignore (hunk 2 above)
Comment (rkostrzewski) [inline .npmignore]: TBH I didn't notice the `files` option. I'll be removing this then

**Pull request merged**